### PR TITLE
[bug fix]refine getTimes and time counting.

### LINKED ISSUE #2499: getTimes() in Container is inconsistent in most of the Containers.
Some container's himself time cost contains submodules' forward/backward time, like StaticGraph.
Some container doesn't return container's time cost, like Concat.
Some container's himself time cost doesn't contain submodules' forward/backward time, like Recurrent.

We need to apply a general rule for these containers, or it's hard to breakdown the time cost in each layer.

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?
Refine time counting:
* Fix time counting doesn't work in some of the modules, like BiRecurrent.
* Fix Cell's `getTimes()` will return a wrong result if call getTimes more than two times.
* Fix `getTimes()` in some Containers doesn't contains himself, like Sequential, Concat.

## How was this patch tested?

unit tests, manual tests on deepspeech2

## Related links or issues (optional)
fixed #2499

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialDivisiveNormalization.scala [hunk 1]
```diff
@@ -137,7 +137,7 @@ class SpatialDivisiveNormalization[T: ClassTag](
   private var localstds: Tensor[T] = _
 
   override def updateOutput(input: Tensor[T]): Tensor[T] = {
-    localstds = stdestimator.updateOutput(input).toTensor[T]
+    localstds = stdestimator.forward(input).toTensor[T]
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialDivisiveNormalization.scala]: you can not get time for each layer in stdestimator

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialSubtractiveNormalization.scala [hunk 2]
```diff
@@ -116,20 +116,20 @@ class SpatialSubtractiveNormalization[T: ClassTag](
       if (dim == 4) {
         // batch mode
         ones.resizeAs(input(1)).fill(ev.fromType[Int](1))
-        val _coef = meanestimator.updateOutput(ones).toTensor[T]
+        val _coef = meanestimator.forward(ones).toTensor[T]
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/SpatialSubtractiveNormalization.scala]: same, you can not get time for each layer in meanestimator

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/TanhShrink.scala [hunk 3]
```diff
@@ -33,7 +33,7 @@ class TanhShrink[T: ClassTag](
   private val tanh = new Tanh[T]()
 
   override def updateOutput(input: Tensor[T]): Tensor[T] = {
-    val th = tanh.updateOutput(input)
+    val th = tanh.forward(input)
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/TanhShrink.scala]: maybe no need, you can not get tanh time

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala [hunk 4]
```diff
@@ -168,6 +168,18 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
     Array((this, forwardTime, backwardTime))
   }
 
+  /**
+   * Get the forward/backward cost time for the module or its submodules
+   * and group by module type.
+   * @return (module type name, forward time, backward time)
+   */
+  final def getTimesGroupByModuleType():
```
Comment (zhangxiaoli73) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala]: when will use this function

Comment (qiuxin2012) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala]: Breaking down the performance in a model.

**Comment by qiuxin2012** [comment]
jenkins passed

**Pull request merged**